test/extensions/network/dns_resolver/getaddrinfo/getaddrinfo_test.cc

```diff
@@ -0,0 +1,292 @@
+#include "envoy/extensions/network/dns_resolver/getaddrinfo/v3/getaddrinfo_dns_resolver.pb.h"
```

Comment:
given cross-thread work let's also make sure one of the e2e tests uses this as well.

Reply:
The tests I wrote does actual resolutions with real threads so it's probably OK? Otherwise I can try to make one of the DNS cache integration tests use this.

Reply:
Looks like no special visibility rules were required so I will add the test.